src/modules/Dashboard/components/TakeNowModal/LabeledDropdown/LabeledUserDropdown.tsx

```diff
@@ -0,0 +1,108 @@
+import unionBy from 'lodash/unionBy';
+import { useTranslation } from 'react-i18next';
+
+import { Tooltip } from 'shared/components';
+import { StyledFlexColumn, StyledTitleMedium, StyledTitleTooltipIcon, theme } from 'shared/styles';
+import { ParticipantSnippet } from 'modules/Dashboard/components/ParticipantSnippet';
+
+import { LabeledUserDropdownProps, ParticipantDropdownOption } from './LabeledUserDropdown.types';
+
+export const LabeledUserDropdown = ({
+  label,
+  name,
+  tooltip,
+  options,
+  value,
+  placeholder,
+  onChange,
+  handleSearch,
+  debounce,
+  ...rest
+}: LabeledUserDropdownProps) => {
+  const { t } = useTranslation('app');
+  const debounceRef = useRef<NodeJS.Timeout | null>(null);
+  const [combinedOptions, setCombinedOptions] = useState<ParticipantDropdownOption[]>(options);
+  const [isSearching, setIsSearching] = useState(false);
+
+  const debouncedSearchHandler = useCallback(
+    (search: string) => {
+      const searchTrimmed = search.trim();
+      if (!handleSearch || !searchTrimmed) {
+        return;
+      }
+
+      if (debounceRef.current) {
+        clearTimeout(debounceRef.current);
+      }
+
+      setIsSearching(true);
+
+      debounceRef.current = setTimeout(async () => {
+        try {
+          let results = handleSearch(searchTrimmed);
+
+          if (results instanceof Promise) {
+            results = await results;
+          }
+
+          setCombinedOptions((prev) => unionBy(prev, results as ParticipantDropdownOption[], 'id'));
+          setIsSearching(false);
+        } catch (_e) {
+          setIsSearching(false);
+        }
+      }, debounce ?? 500);
+    },
+    [debounce, handleSearch],
+  );
+
+  return (
+    <StyledFlexColumn sx={{ gap: 1.6, marginTop: 2.4 }}>
+      <Box sx={{ display: 'flex', gap: 0.4 }}>
+        <StyledTitleMedium fontWeight="bold">{label}</StyledTitleMedium>
+        <Tooltip tooltipTitle={tooltip}>
+          <Box sx={{ height: 24 }}>
+            <StyledTitleTooltipIcon
+              sx={{ marginLeft: 0 }}
+              id="more-info-outlined"
+              width={24}
+              height={24}
+              data-testid={`${rest['data-testid']}-tooltip-icon`}
+            />
+          </Box>
+        </Tooltip>
+      </Box>
+      <Autocomplete
+        renderInput={(params) => {
+          const { InputLabelProps: _InputLabelProps, ...rest } = params;
+
+          return <TextField {...rest} placeholder={placeholder} name={name} />;
+        }}
+        options={combinedOptions}
+        renderOption={({ children: _children, ...props }, { id: _id, ...psProps }) => (
+          <ParticipantSnippet<'li'>
+            {...psProps}
+            boxProps={{
+              sx: { p: theme.spacing(0.6, 1.6), cursor: 'pointer' },
+              ...props,
+            }}
+          />
+        )}
+        getOptionLabel={(value) =>
+          `${value.secretId}${value.nickname ? ` (${value.nickname})` : ''}`
```

Comment:
Would the effort to allow `Autocomplete` to accept a `<ParticipantSnippet>` for its currently selected value be too much for this PR? I see that there are probably some structural/UX considerations; i.e. presumably it would only show when focus is blurred because the user has to be able to access the (plain-text) search input field when focused.

Reply:
I should've mentioned this. I didn't see a straightforward way to implement it without doing something fancy like you described (especially since Matt stressed that we should try to reuse existing component as much as possible),and I didn't want to drag out the PR any more, so I fell back to just using plain text.

I'd love to get this matching the design eventually though